### DISCUSSION TITLE
Cannot verify a smart contract on Etherscan

### DISCUSSION THREAD

**Discussion by 0xdavinci** [Cyfrin/foundry-full-course-cu#2748] [Q&A]
I deployed the smart contract on Sepolia.
My contract used solidity 0.8.28 .
Here is my contract address.
https://sepolia.etherscan.io/address/0x69be1adb241e9215633f48fee8d7532da77c83d0
I tried to verify and publish my contract source code.
I set the Compiler Version to v0.8.28+commit.7893614a.
![Screenshot 2024-10-24 220437](https://github.com/user-attachments/assets/a98a7c52-a4a2-4b56-bf27-8acc1a4ff2c1)

Error! Unable to generate Contract Bytecode and ABI

**Comment by EngrPips** [comment]
Hello @0xdavinci, I always verify my contract the moment I deploy using the foundry `--verify` flag, as I wasn't successful when I tried to verify an already deployed contract on etherscan UI. I trust someone that know what could be wrong will see this and respond as soon as possible.

**Reply by EdwinFairchild** [reply]
how do you add that flag in remix ide?

**Comment by DCVglobalnetwork** [comment]
Hi 
I had similar experience. Once i did verify, once i did not 
any suggestions?

**Reply by EngrPips** [reply]
I am not sure what the best suggestion will be but verifying at deployment time have always worked for me.

**Comment by KMean** [comment]
@0xdavinci @EdwinFairchild @DCVglobalnetwork  If you want to verify you contract after has been deployed, you need to use a plugin on Remix called contract verification.

![verifycontract](https://github.com/user-attachments/assets/87c76fbb-311d-4915-84eb-39176634115e)
![contractverification](https://github.com/user-attachments/assets/5bebe79f-61e1-4ac2-9a11-6b48cebecbf0)

to use it on etherscan you need your etherscan API key otherwise you can use blockscout.
If you're in forge, provided you have your etherscan api key, simply use the --verify flag as @EngrPips suggested or use one of the methods described in [the foundry book](https://book.getfoundry.sh/reference/forge/forge-verify-contract)

**Reply by EngrPips** [reply]
Great heads-up.

**Reply by DCVglobalnetwork** [reply]
@KMean KMean yes thank you for your help I also figured out that my compiler version did not match the one I deployed. So always has to match the compiler vision with deployed one.

**Reply by EngrPips** [reply]
Interesting.

**Reply by EdwinFairchild** [reply]
Thanks for the reply!!
